fabric-fluids-v1/src/main/java/net/fabricmc/fabric/api/fluids/containers/SimpleFluidContainer.java

```diff
@@ -0,0 +1,113 @@
+package net.fabricmc.fabric.api.fluids.containers;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+import com.google.common.collect.Iterables;
+
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.fluids.containers.volume.FluidVolume;
+import net.fabricmc.fabric.api.fluids.containers.volume.SimpleFluidVolume;
+
+public class SimpleFluidContainer implements FluidContainer {
+	private final Iterable<FluidContainer> containers;
+
+	public SimpleFluidContainer(FluidContainer... containers) {
+		this(Arrays.asList(containers));
+	}
+
+	public SimpleFluidContainer(Iterable<FluidContainer> containers) {
+		this.containers = containers;
+	}
+
+	@Override
+	public FluidVolume drain(Identifier fluid, long amount, boolean simulate) {
+		FluidVolume current = new SimpleFluidVolume();
+		for (FluidContainer container : this.containers) {
+			FluidVolume drained = container.drain(current.isEmpty() ? fluid : current.fluid(), amount, false);
+			current.add(drained, false);
+			amount -= drained.amount();
+			if (amount <= 0) {
+				break;
+			}
+		}
+		return current;
+	}
+
+	@Override
+	public FluidVolume add(FluidVolume volume, boolean simulate) {
+		for (FluidContainer container : this.containers) {
+			volume = container.add(volume, simulate);
+			if (volume.isEmpty()) {
+				break;
+			}
+		}
+		return volume;
+	}
+
+	@Override
+	public Iterator<FluidVolume> iterator() {
+		Iterator<FluidContainer> containers = this.containers.iterator();
+		return new Iterator<FluidVolume>() {
```

Comment:
there has to be a better way to do this Iterator stuff?

Reply:
I spent way too long fiddling with the Iterators and Iterables classes, I'll have another go at it

Reply:
turns out, I'm an idiot, found a solution